zipkin-lens/src/components/TracePage/TraceSummaryHeader.jsx

```diff
@@ -32,84 +35,135 @@ const defaultProps = {
+              {entry.value}
+            </Box>
+          </Box>
+        ))
+      }
+    </Box>
+  ) : <div />;
+
   return (
-    <Box pl={3} pr={3}>
-      <Box
-        width="100%"
-        display="flex"
-        justifyContent="space-between"
-        borderBottom={1}
-        className={classes.headerTop}
-      >
-        <Box display="flex" alignItems="center">
+    <Box className={classes.root}>
+      <Box className={classes.upperBox}>
+        <Box className={classes.serviceNameAndSpanName}>
           {
             traceSummary ? (
               <>
-                <Box className={classes.serviceName}>
-                  <Typography variant="h5">
-                    {traceSummary.rootSpan.serviceName}
-                  </Typography>
-                </Box>
-                <Box className={classes.spanName}>
-                  <Typography variant="h5">
-                    {`: ${traceSummary.rootSpan.spanName}`}
-                  </Typography>
-                </Box>
+                <Typography variant="h5" className={classes.serviceName}>
+                  {traceSummary.rootSpan.serviceName}
+                </Typography>
+                <Typography variant="h5" className={classes.spanName}>
+                  {` : ${traceSummary.rootSpan.spanName}`}
+                </Typography>
               </>
             ) : null
           }
         </Box>
-        <Box pr={4} display="flex" alignItems="center">
+        <Box className={classes.jsonUploaderAndSearchInput}>
           <TraceJsonUploader />
           <TraceIdSearchInput />
         </Box>
       </Box>
-      <Box
-        display="flex"
-        mt={0.5}
-        mb={0.5}
-        alignItems="center"
-        fontSize="1rem"
-      >
-        {
-          traceSummary ? (
-            [
-              { label: 'Duration', value: traceSummary.durationStr },
-              { label: 'Services', value: traceSummary.serviceNameAndSpanCounts.length },
-              { label: 'Depth', value: traceSummary.depth },
-              { label: 'Total Spans', value: traceSummary.spans.length },
-              {
-                label: 'Trace ID',
-                value: rootSpanIndex === 0
-                  ? traceSummary.traceId
-                  : `${traceSummary.traceId} - ${traceSummary.spans[rootSpanIndex].spanId}`,
-              },
-            ].map(e => (
-              <Box key={e.label} mr={1} display="flex">
-                <Box fontWeight="bold" color="grey.600">
-                  {`${e.label}:`}
-                </Box>
-                <Box fontWeight="bold" ml={0.8}>
-                  {e.value}
-                </Box>
-              </Box>
-            ))
-          ) : null
-        }
+      <Box className={classes.lowerBox}>
+        {traceInfo}
+        <Button variant="outlined" className={classes.saveButton} onClick={handleSaveButtonClick}>
+          <Box component="span" className={`${classes.saveButtonIcon} fas fa-download`} />
+          Save JSON
```

Comment:
Just a nit pick on the text of the button. I think "Download JSON" may be better because the trace is technically already saved in the backend. Maybe others won't have any confusion of these terms. Anyone else have an opinion?